ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/ExecutionEnvironmentSection.java

```diff
@@ -311,7 +329,15 @@ private void handleRemove() {
 	}
 
 	private void handleAdd() {
-		ElementListSelectionDialog dialog = new ElementListSelectionDialog(PDEPlugin.getActiveWorkbenchShell(), new EELabelProvider());
+		ElementListSelectionDialog dialog = new ElementListSelectionDialog(PDEPlugin.getActiveWorkbenchShell(),
+				new EELabelProvider()) {
+			@Override
+			protected FilteredList createFilteredList(Composite parent) {
+				FilteredList filteredList = super.createFilteredList(parent);
+				filteredList.setComparator(new VersionComparator());
```

Comment:
As mentioned below the EEs are probably already sorted in the EE-Manager. So just reversing the list might be sufficient.

Reply:
@HannesWell  Yes. They are sorted based on string comparison and not versions. 
If I am not wrong the list is sorted in natural order in `setElements` method of `FilteredList` class (unless we set a comparator explicitly ) even if the list is revered before setting it to dialog.